debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java

```diff
@@ -379,9 +379,17 @@ public OracleConnectorConfig(Configuration config) {
         this.logMiningHistoryRecorder = resolveLogMiningHistoryRecorder(config);
         this.jdbcConfig = config.subset(DATABASE_CONFIG_PREFIX, true);
         this.snapshotEnhancementToken = config.getString(SNAPSHOT_ENHANCEMENT_TOKEN);
+        this.connectorAdapter = ConnectorAdapter.parse(config.getString(CONNECTOR_ADAPTER));
```

Comment:
Could you use the code from `io.debezium.config.Configuration.getInstance(Field, Class<T>, Supplier<ClassLoader>)`?

Reply:
I could, but probably not in the same way in which we use this for things such as `database.history` in MySQL.  The property that is being used here is an Enum and so what we'd need to do is actually directly call `Instantiator` directly with the right arguments much like what is done in the `Configuration` interface.  I'll provide a commit to illustrate this and you can let me know if it's acceptable.

Reply:
See https://github.com/debezium/debezium/pull/2274/commits/d5851862515b6ac8d3bae971183a10bdd7a9e5d7

Reply:
That's exactly what I had in my mind